tools/config.py

```diff
@@ -797,6 +798,13 @@ def validate_config(self):
         return True
 
 
+    @property
+    def name(self):
+        if "artifact_name" in self.app_config_data:
```

Comment:
I would suggest only doing this for apps for now. Not every `mbed_lib.json` corresponds to a an archived library build. In fact, it almost never works like this. At least you know there is only one app, so you can lock down the name of the produced binary.